apps/docs/content/guides/auth/third-party/firebase-auth.mdx

```diff
@@ -0,0 +1,254 @@
+
+<Admonition type="caution">
+
+Postgres has two types of policies: permissive and restrictive. This example uses restrictive policies so make sure you don't omit the `as restrictive` clause.
+
+</Admonition>
+
+This is an example of such an RLS policy that will restrict access to only your project's (denoted with `<firebase-project-id>`) users, and not any other Firebase project.
+
+```sql
+create policy "Restrict access to Supabase Auth and Firebase Auth for project ID <firebase-project-id>"
+  on table_name
+  as restrictive
+  to authenticated
+  using (
+    (auth.jwt()->>'iss' = 'https://<project-ref>.supabase.co/auth/v1')
+    or
+    (
+        auth.jwt()->>'iss' = 'https://securetoken.google.com/<firebase-project-id>'
+        and
+        auth.jwt()->>'aud' = '<firebase-project-id>'
+     )
+  );
+```
+
+If you have a lot of tables in your app, or need to manage complex RLS policies for [Storage]() or [Realtime]() it can be useful to define a [stable Postgres function](https://www.postgresql.org/docs/current/xfunc-volatility.html) that performs the check to cut down on duplicate code. For example:
+
+```sql
+create function public.is_supabase_or_firebase_project_jwt()
+  returns bool
+  language sql
+  stable
+  returns null on null input
+  return (
+    (auth.jwt()->>'iss' = 'https://<project-ref>.supabase.co/auth/v1')
+    or
+    (
+        auth.jwt()->>'iss' = concat('https://securetoken.google.com/<firebase-project-id>')
+        and
+        auth.jwt()->>'aud' = '<firebase-project-id>'
+     )
+  );
+```
+
+Please make sure you substitute `<project-ref>` with your Supabase project's ID and the `<firebase-project-id>` to your Firebase Project ID. Then the restrictive policies on all your tables, buckets and channels can be simplified to be:
+
+```sql
+create policy "Restrict access to correct Supabase and Firebase projects"
+  on table_name
+  as restrictive
+  to authenticated
+  using (public.is_supabase_or_firebase_project_jwt() is true);
+```
+
+## Assign the "role" custom claim
+
+Your Supabase project inspects the `role` claim present in all JWTs sent to it, to assign the correct Postgres role when using the Data API, Storage or Realtime authorization.
+
+By default, Firebase JWTs do not contain a `role` claim in them. If you were to send such a JWT to your Supabase project, the `anon` role would be assigned when executing the Postgres query. Most of your app's logic will be accessible by the `authenticated` role.
+
+The easiest way to achieve this is to define a [blocking Firebase Authentication function](https://firebase.google.com/docs/auth/extend-with-blocking-functions). However this is only available if your Firebase project uses [Firebase Authentication with Identity Platform](https://cloud.google.com/identity-platform/docs/product-comparison).
```

Comment:
we could link to the sub-sections below instead of linking to the firebase docs here. to make it very explicit that there are 2 ways and they are described in the sections below. we can link to the firebase docs from the subsections instead so that the flow is clear